Fix #1136: heatmap and toolbox hide/highlight need x and y to be swapped

sergpolly opened the pull request:
issue descibed in #1136 seem to originate from the x/y swap - i.e. `xcats` and `ycats` of the heatmap (rows and columns respectively) are referring to wrong indexes in the sparse-matrix variable `data` that stores all of the heatmap's data: `x` refers to `data[n][1]` and `y` refer to `data[n][0]`, but it should be `x` <-->`data[n][0]` and `y` <--> `data[n][1]` instead.

`data` - seem to be a sprase matrix in COO format and it seem to follow the `[row, col, value]` layout
![Screenshot from 2020-03-28 22-56-50](https://user-images.githubusercontent.com/6790270/77839765-bcab8800-714d-11ea-8f0a-c46103593f38.png)

In this PR i'll carefully flip/swap `x/y` `0/1` when needed , and add some comments to hopefully ease community-maintenance.

Just one commit for now, fixing the `hide` aspect of the toolbox
[to be continued]

## If this PR is _not_ a new module
 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` has been updated
 - [x] (optional but recommended): https://github.com/ewels/MultiQC_TestData contains test data for this change

ewels commented:
This is fab, thanks @sergpolly - good detective work! I've labelled this PR as a work in progress for now. Let me know when the `console.log` statements are gone and it's good to merge.. 👍

sergpolly commented:
hey @ewels - I think this one is done:
 - I completed partial x/y <-> 0/1 swap fix (partial because it was correct in some places - e.g. "sort by highlight")
 - I removed debug-related code, minimized `diffs` but decided to keep some comments there ... 
 - updated `changelog`

here is the demo after fixing:
![hide_highligh](https://user-images.githubusercontent.com/6790270/78082614-776f9c00-7381-11ea-89a0-2c8998de1114.gif)

all seemed great! ....

PS
working on this I realized another potential issue with applying toolbox to the heatmaps ...
right now `xcats` and `ycats` in the `heatmap` code are treated absolutely symmetrically - i.e. `rename/hide/highlight` are applied to both `xcats` and `ycats` regardless of the context. This should work as intended for a square sample-to-sample similarity/correlation matrix/heatmap. For other cases, this, of course, can lead to some unintended interference between sample names and the other category names (the non-sample one), although arguably it is rare scenario. The issue becomes absolutely *screaming*, when one chooses to `hide` in `Show only matching sa mples`-regime - see another demo:
![hide_show_matching](https://user-images.githubusercontent.com/6790270/78083861-b6532100-7384-11ea-9c03-f0dd280b81e8.gif)
`SRR106` matches a bunch of samples but none of the `xcats` - column labels - which makes `heatmap` yield "blank" - should I open another issue to keep this in the record ?

ewels commented:
> `SRR106` matches a bunch of samples but none of the `xcats` - column labels - which makes `heatmap` yield "blank" - should I open another issue to keep this in the record ?

Ah pants - yes that's annoying. A new issue would be fab 👍 I'm not really sure how we address that... 🤔  Maybe have different behaviour depending on whether the heatmap is square or not?

ewels commented:
This is great - tested and working really well in my hands. Thanks for this!
